src/upkeep.rs

```diff
@@ -480,8 +430,11 @@ mod tests {
         batch[2].remove_at = Utc.with_ymd_and_hms(2024, 11, 14, 21, 22, 23).unwrap();
 
         assert!(store.store(batch.clone()).await.is_ok());
-        do_upkeep(config, store.clone(), producer).await;
+        let result_context = do_upkeep(config, store.clone(), producer).await;
 
+        assert_eq!(result_context.remove_at_expired, 1); // batch[2] is removed due to remove_at deadline
+        assert_eq!(result_context.discarded, 1); // batch[2] is discarded
+        assert_eq!(result_context.completed, 2); // batch[1] and batch[2] are removed as completed
```

Comment:
I think the comment on 477 is also incorrect. It is index 0 that would be discarded, not 2. The assertions for row presence on 493 are also not right. 

```rust
        assert!(
            store.get_by_id(&batch[0].activation.id).await.is_ok(),
            "first task should remain"
        );
```

Should be 

```rust
        assert!(
            store.get_by_id(&batch[0].activation.id).await.unwrap().is_none(),
            "first task should be removed"
        );
        assert!(
            store.get_by_id(&batch[1].activation.id).await.unwrap().is_none(),
            "second task should be removed"
        );
        assert!(
            store.get_by_id(&batch[2].activation.id).await.unwrap().is_some(),
            "third task should be kept"
        );
```

Reply:
Good catch, I had the comment the other way around. Also updated the assertion for row presence.